src/server/svr_func.c

```diff
@@ -5236,7 +5239,24 @@ is_vnode_prov_done(char * vnode)
 			id, prov_vnode_info->pvnfo_vnode))
 		return;
 	}
+	
+	/* check if IS_HOOK_CHECKSUMS is processed if not then
+ 	 * come back after 5 sec and check again, this affirms
+ 	 * the hook files are in sync before starting the job */
 
+	if (!((mom_svrinfo_t *)(pnode->nd_moms[0]->mi_data))->msr_is_hook_checksums_done
+			&& !sync_mom_hookfiles_proc_running && !do_sync_mom_hookfiles){
```

Comment:
Hi Dilip,
I am sorry I misinterpreted your previous comment. Why I added this code in is_vnode_prov_done() is, from time task is_sync_hooks_done() I am checking node state for INUSE_PROV before calling is_vnode_prov_done() and this will get reset before calling is_runnable().

Regards,
Riyaz